### ISSUE TITLE
"old tweets" and date ranges apparently don't work together

### ISSUE THREAD

**Issue by ex-nerd** [Lyfhael/DeleteTweets#17]
I was having trouble getting tweets before 2023 to delete, so enabled `old_tweets` along with my `before_date=2023-01-01`. After the script ran, pretty much all of my tweets/replies/retweets were gone (it somehow missed a few from the last couple weeks).

**Comment by Lyfhael** [comment]
I can't seem to reproduce this error. I tried old_tweets=true and a date_range that excludes some of my tweets, and these were effectively not deleted.

So there may be more specific conditions inducing but I can't guess which ones, I would need to be able to reproduce the error and check how the response json is formatted

**Comment by ex-nerd** [comment]
Yeah, I wish I had a way to try again for more debugging, but … well, all of my tweets are gone. 🤷‍♂️

No biggie, thought it would be worth looking into if it was something obvious.

**Comment by Lyfhael** [comment]
> Yeah, I wish I had a way to try again for more debugging, but … well, all of my tweets are gone. 🤷‍♂️
> 
> No biggie, thought it would be worth looking into if it was something obvious.

My guess would be that the dates are formatted differently in old tweets (Which I couldn't test because despite enable "old_tweets" I simply didn't have any old tweets so the script just deleted the recent ones)

If you know or have another account with old tweets and someone willing to remove them, we may find what the issue is

**Comment by Lyfhael** [comment]
Hey, just letting you know you can now use your Twitter archive to delete your tweets, it should circumvent this issue ;)

**Comment by roshanmirajkar** [comment]
I also had trouble retrieving info before 2023. Then I tried the _delete_option_ of using Twitter archive. Works more seamless than 90% of tweet deletion tools I've tried. Thanks Lyfhael!

**Comment by hebamuh68** [comment]
@Lyfhael how u use Twitter archive to delete all tweets?
